### PR TITLE
feat(op-challenger): Validate Step Function Manual ABI Packing

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Validates the step function packing as a unit test in the fault responder.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6258** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6258" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6258?utm_source=stack-comment).

File: op-challenger/fault/abi_test.go [hunk 1]
```diff
@@ -72,3 +72,26 @@ func TestBuildFaultAttackData(t *testing.T) {
...
+// TestBuildFaultStepData ensures that the manual ABI packing is the same as going through the bound contract.
+func TestBuildFaultStepData(t *testing.T) {
+	_, opts, _, contract, err := setupFaultDisputeGame()
+	require.NoError(t, err)
+
+	responder, _ := newTestFaultResponder(t, false)
+
+	data, err := responder.buildStepTxData(StepCallData{
+		StateIndex: 1,
+		ClaimIndex: 2,
```
Comment (trianglesphere) [inline op-challenger/fault/abi_test.go:84]: `StepCallData` needs to be changed b/c `ClaimIndex` no longer exists

Comment (refcell) [reply op-challenger/fault/abi_test.go:84]: The `ClaimIndex` does exist though in the `step` function call in the FaultDisputeGame contract (see [here](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/dispute/FaultDisputeGame.sol#L74)).

Just pushed a change to remove the `StateIndex` and keep the `ClaimIndex`. I also left a [comment](https://github.com/ethereum-optimism/optimism/pull/6256/files#r1260030145) in your draft pr addressing the removal of the `ClaimIndex` and leaving the `StateIndex`.

**Review by trianglesphere — APPROVED** [review]
Lgtm

**Pull request merged**